packages/api/tests/lambdas/test-cleanExecutions.js

```diff
@@ -1,58 +1,414 @@
 
-const { cleanExecutionPayloads } = require('../../lambdas/cleanExecutions');
-
-const removeRecordsStub = sinon.spy();
+  t.context.knex = knex;
+  t.context.knexAdmin = knexAdmin;
+  const { esIndex, esClient, searchClient } = await createTestIndex();
+  t.context.esIndex = esIndex;
+  t.context.esClient = esClient;
+  t.context.searchClient = searchClient;
+  const records = [];
+  for (let i = 0; i < 10; i += 1) {
+    records.push(fakeExecutionRecordFactory({
+      updated_at: moment().subtract(i, 'days').toDate(),
+      final_payload: '{"a": "b"}',
+      original_payload: '{"b": "c"}',
+      status: 'running',
+    }));
+    records.push(fakeExecutionRecordFactory({
+      updated_at: moment().subtract(i, 'days').toDate(),
+      final_payload: '{"a": "b"}',
+      original_payload: '{"b": "c"}',
+      status: 'failed',
+    }));
+  }
+  const model = new ExecutionPgModel();
 
-class executionModel {
-  removeOldPayloadRecords(...value) {
-    return removeRecordsStub(...value);
+  const pgRecords = await model.insert(t.context.knex, records, '*');
+  for (const record of pgRecords) {
+    await t.context.esClient.client.index({
+      body: await translatePostgresExecutionToApiExecution(record),
+      id: record.cumulus_id,
+      index: t.context.esIndex,
+      type: 'execution',
+      refresh: true,
+    });
   }
-}
+  t.context.execution_cumulus_ids = pgRecords.map((record) => record.cumulus_id);
+});
 
-test.beforeEach(() => {
-  process.env.completeExecutionPayloadTimeoutDisable = false;
-  process.env.nonCompleteExecutionPayloadTimeoutDisable = false;
+test.afterEach.always(async (t) => {
+  await destroyLocalTestDb({
+    knex: t.context.knex,
+    knexAdmin: t.context.knexAdmin,
+    tesetDbName: t.context.testDbName,
+  });
+  await cleanupTestIndex(t.context);
 });
 
-test.serial('Function is called with correct timeout values', async (t) => {
-  process.env.completeExecutionPayloadTimeout = 100;
-  process.env.nonCompleteExecutionPayloadTimeout = 50;
-  await cleanExecutionPayloads(executionModel);
-  t.truthy(removeRecordsStub.calledWith(100, 50, false, false));
+const pgPayloadsEmpty = (entry) => !entry.final_payload && !entry.orginal_payload;
+
+const esPayloadsEmpty = (entry) => !entry.finalPayload && !entry.orginalPayload;
+
+test.serial('handler() handles running expiration', async (t) => {
+  const env = clone(process.env);
+  process.env = localStackConnectionEnv;
+  process.env.PG_DATABASE = t.context.testDbName;
+  process.env.ES_INDEX = t.context.esIndex;
+  process.env.LOCAL_ES_HOST = 'localhost';
+  let expirationDays = 4;
+  let expirationDate = getExpirationDate(expirationDays);
+  process.env.CLEANUP_NON_RUNNING = 'false';
+  process.env.CLEANUP_RUNNING = 'true';
+  process.env.CLEANUP_POSTGRES = 'true';
+  process.env.CLEANUP_ES = 'true';
+  process.env.PAYLOAD_TIMEOUT = expirationDays;
+
+  await handler();
+  await sleep(5000);
```

Comment:
This may have been covered before but why the sleep here? Asking about these because I'm worried they're fragile.

Reply:
Also as a serial test, this will slow down the units by 5 seconds. Not the end of the world but...

Reply:
this is a relic of previous paradigm:
the reason *was* that it's super slow to hard refresh the elasticsearch index when calling es operations multiple times, and so this was a way to test effectively without forcing index refreshes
*but* we're only making one es op call so it's trivial to refresh the index afterwards

Reply:
https://github.com/nasa/cumulus/pull/3721/commits/e8f221b21ffc58b85da35873f9862feb4bbedc67